### PR TITLE
Fixes #16886: rudder-lang packaging update

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16886

no module dependencies left so rudder-lang should be easy to integrate now

### CODE REVIEW

File: rudder-lang/Makefile [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
+
+	# Install executable and helper scripts
+	install -m 755 target/release/rudderc $(DESTDIR)/opt/rudder/bin/rudderc
+	install -m 755 tools/tester.sh $(DESTDIR)/opt/rudder/share/rudder-lang/tools/tester.sh
+	install -m 755 tools/cfjson_tester $(DESTDIR)/opt/rudder/share/rudder-lang/tools/cfjson_tester
+	install -m 755 tools/generate_lib $(DESTDIR)/opt/rudder/share/rudder-lang/tools/generate_lib
+	
+	# Copy configuration and library files
+	# required for rudderc execution
+	install -m 644 tools/rudderc.conf $(DESTDIR)/opt/rudder/etc/rudderc.conf
```
Comment (peckpeck) [inline rudder-lang/Makefile]: configuration should be 640

File: rudder-lang/Makefile [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
+	install -m 644 tools/docopt.py $(DESTDIR)/opt/rudder/share/python/docopt.py
+	install -m 644 libs/corelib.rl $(DESTDIR)/opt/rudder/share/rudder-lang/lib/corelib.rl
+	install -m 644 libs/cfengine_core.rl $(DESTDIR)/opt/rudder/share/rudder-lang/lib/cfengine_core.rl
+	install -m 644 libs/oslib.rl $(DESTDIR)/opt/rudder/share/rudder-lang/lib/oslib.rl
+	cp -R tools/jsondiff $(DESTDIR)/opt/rudder/share/python/ && chmod 755 $(DESTDIR)/opt/rudder/share/python/jsondiff && chmod 644 $(DESTDIR)/opt/rudder/share/python/jsondiff/*
```
Comment (peckpeck) [inline rudder-lang/Makefile]: usually we say -r

File: rudder-lang/src/translate.rs [hunk 3]
```diff
@@ -393,7 +393,8 @@ fn translate_condition(_config: &toml::Value, cond: &str) -> Result<String> {
         // matches a word that can be negative and that be followed by .| + word... possibly wrapped into parenthesis
         static ref OS_RE: Regex = Regex::new(
             // OS part: debian_9_0 \ And optional condition: (!(ubuntu|otheros).something)
-            r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            // r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            r"^(\(*\w+\(*\)*(\||\.)\(*\w+\)*)*$"
```
Comment (peckpeck) [inline rudder-lang/src/translate.rs]: || should not appear in a regex, either you mean 
- a litteral || : with \|\|
- an OR operation : with |

Comment (gpoblon) [reply rudder-lang/src/translate.rs]: \||\. => I meant an Or operation : must found either a "|" or "."
What should I do then ?

Comment (gpoblon) [reply rudder-lang/src/translate.rs]: I could rewrite it like this to clarify: `(\\.|\\|)`

**Comment by gpoblon** [comment]
Applied requested changes

File: rudder-lang/src/translate.rs [hunk 4]
```diff
@@ -393,7 +393,8 @@ fn translate_condition(_config: &toml::Value, cond: &str) -> Result<String> {
         // matches a word that can be negative and that be followed by .| + word... possibly wrapped into parenthesis
         static ref OS_RE: Regex = Regex::new(
             // OS part: debian_9_0 \ And optional condition: (!(ubuntu|otheros).something)
-            r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            // r"^\(?(?P<os>([a-zA-Z\d]+)(_([a-zA-Z\d]+))*(\|([a-zA-Z\d]+)(_([a-zA-Z\d]+))*)*)\)?(.(?P<cdt>\(\(*!*\(*\w+\)*([.|]\(*!*\(*\w+\)*)*\)))?$"
+            r"^(\(*\w+\(*\)*(\.|\|)\(*\w+\)*)*$"
```
Comment (peckpeck) [inline rudder-lang/src/translate.rs]: (\.|\|) qui equivalent to [.|] which is easier to read

Comment (peckpeck) [reply rudder-lang/src/translate.rs]: this regex looks dubious \(*  but it will do for a first step and will probably need further improvements

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**